### ISSUE TITLE
initrock should drop rockstor specific databases rather than del dir

### ISSUE THREAD

**Issue by phillxnet** [rockstor/rockstor-core#2076]
During a source build, or a source to rpm transition, initrock does the python equivalent of 'rm -rf /var/lib/pgsql/data' this has been known to cause postgresql to coredump. Restart was auto enacted but a more 'measured' approach of dropping our known databases only would be more appropriate and likely more predictable.

**Comment by phillxnet** [comment]
In addition to the above, it looks like we may have an ordering issue at least on a clean source build for systems that already have postgresql running:
```
[02/May/2020 10:16:53] DEBUG [system.osi:151] RUN_COMMAND udevadm info --name /dev/vda
[02/May/2020 10:16:53] DEBUG [system.osi:151] RUN_COMMAND /usr/bin/systemctl enable postgresql
[02/May/2020 10:16:53] DEBUG [scripts.initrock:418] Deleting /var/lib/pgsql/data
[02/May/2020 10:16:53] DEBUG [scripts.initrock:429] running generic initdb on /var/lib/pgsql/data
[02/May/2020 10:16:53] DEBUG [system.osi:151] RUN_COMMAND su - postgres -c /usr/bin/initdb -D /var/lib/pgsql/data
[02/May/2020 10:16:58] DEBUG [system.osi:151] RUN_COMMAND /usr/bin/systemctl restart postgresql
[02/May/2020 10:16:58] DEBUG [system.osi:151] RUN_COMMAND /usr/bin/systemctl status postgresql
[02/May/2020 10:16:58] DEBUG [system.osi:151] RUN_COMMAND su - postgres -c /usr/bin/createdb smartdb
```
It would be prudent to explicitly stop postgresql prior to deleting the db dir, however the issue text approach would be much preferred, ie dropping our known databases only rather than taking a filesystem level approach.

**Comment by phillxnet** [comment]
Note that the Postresql service failure detailed in the later (marked as duplicate) issue of #2361 can be recreated using the following bash script:

```
# Executed on a system that has a prior Rockstor db setup:
# i.e. rockstor-pre (initrock) has previously instantiated a prior Rockstor db setup.
#
# Perform a db reset via initrock where it rudely deletes from under Postresql
systemctl stop rockstor rockstor-pre rockstor-bootstrap
rm /opt/rockstor-dev/.initrock
#
# The following initiates rockstor-pre (initrock) which deletes Postresql files!!
# Because it finds no .initrock 'flag' file.
systemctl start rockstor

# Check status of our critical servcies.
systemctl --no-pager status postgresql
systemctl --no-pager status rockstor-pre
```

I.e. we simply invoke the at-question db reset mechanism currently employed by initrock, resulting in a broken postresql scenario which then requires the 'clean-up' procedure descried in the comments of the duplicate issue #2361. Ergo we have an observable/reproducible fault caused by our inelegance regarding existing (though prior instance) database treatment on cleaning house. That is we do it from under Postresql, rather than requesting the same from Postresql itself. Thus avoiding the likely need to manually intervene in an otherwise dysfunctional critical service scenario.

**Comment by phillxnet** [comment]
Given we are now in testing branch modifications, we can assume postgresql version 13 or greater (See https://github.com/rockstor/rockstor-core/pull/2407#issuecomment-1300879372) in the now merged in testing branch changes that instigated the change form posgresql10-server to  postgresql13-server in our related packaging dependencies.

The proposed (postresql13) command to wipe our existing database is as follows:
```
dropdb --force --if-exists db-name-here
```
Note that the --force option (our preferred) is only in version 13 and onwards of postgresql.

**Comment by phillxnet** [comment]
N.B. our underlying OS may well now have both postgres10 and 13 installed. The default (auto) for alternatives auto adjusts paths to use 13 (priority 130) to 10 (priority 100). See:
```
update-alternatives --display postgresql
```

**Comment by phillxnet** [comment]
Closing as:
Fixed by #2420
